### PR TITLE
Feature/extra warning messages on academic interfaces FIST-441 #resolve

### PR DESCRIPTION
**Pull request by Luis-Cruz**

### CODE REVIEW

File: fenixedu-ist-giaf-invoices/src/main/java/pt/ist/fenixedu/giaf/invoices/Utils.java [hunk 1]
```diff
@@ -509,4 +512,37 @@ public static void writeFileWithoutFailuer(final Path path, final byte[] content
...
+
+    private static DateTime getDueDateByPaymentCodes(final Event event) {
+        final YearMonthDay ymd =
+                event.getPaymentCodesSet().stream().map(pc -> pc.getEndDate()).max((c1, c2) -> c1.compareTo(c2)).orElse(null);
+        return ymd != null ? ymd.plusDays(1).toDateTimeAtMidnight() : event.getWhenOccured();
+    }
+
+    private static DateTime findLastDueDate(final GratuityEventWithPaymentPlan event) {
+        return event.getInstallments().stream().map(i -> i.getEndDate().toDateTimeAtMidnight()).max(new Comparator<DateTime>() {
+            @Override
```
Comment (duchuyctlk) [inline fenixedu-ist-giaf-invoices/src/main/java/pt/ist/fenixedu/giaf/invoices/Utils.java]: Could you tell why don't you use lambda as above?

Comment (Luis-Cruz) [inline fenixedu-ist-giaf-invoices/src/main/java/pt/ist/fenixedu/giaf/invoices/Utils.java]: Should be better now.
Thanks for the observation.

File: fenixedu-ist-integration/src/main/java/pt/ist/fenixedu/integration/ui/FenixEduISTLegacyContextListener.java [hunk 2]
```diff
@@ -139,6 +151,77 @@ public String uVATNumberFor(final Person person) {
...
+
+        StudentWarningsService.register(new Function<Student, Collection<String>>() {
+
+            @Override
+            public Collection<String> apply(final Student student) {
+                final Collection<String> warnings = new TreeSet<String>();
+
+                final Person person = student.getPerson();
+                final Country country = person.getCountry();
+                if (person.getCountry() == null) {
```
Comment (duchuyctlk) [inline fenixedu-ist-integration/src/main/java/pt/ist/fenixedu/integration/ui/FenixEduISTLegacyContextListener.java]: IMO `country` could be used here.

**Comment by sergiofbsilva** [comment]
@duchuyctlk do you have any other questions, or can we accept this ? 👍

**Pull request merged**